covalent_dispatcher/_core/execution.py

```diff
@@ -582,47 +631,58 @@ def task_callback(future: Future):
         wait(futures)
         # del futures
 
-        # When one or more nodes failed in the last iteration, don't iterate further
-        for node_id in nodes:
-            if result_object._get_node_status(node_id) == Result.FAILED:
-                result_object._status = Result.FAILED
-                result_object._end_time = datetime.now(timezone.utc)
-                result_object._error = f"Node {result_object._get_node_name(node_id)} failed: \n{result_object._get_node_error(node_id)}"
-                with DispatchDB() as db:
-                    db.upsert(result_object.dispatch_id, result_object)
-                    db.save_db(result_object)
-                result_webhook.send_update(result_object)
-                return result_object
-
-            elif result_object._get_node_status(node_id) == Result.CANCELLED:
-                result_object._status = Result.CANCELLED
-                result_object._end_time = datetime.now(timezone.utc)
-                with DispatchDB() as db:
-                    db.upsert(result_object.dispatch_id, result_object)
-                    db.save_db(result_object)
-                result_webhook.send_update(result_object)
-                return result_object
-
-    # post process the lattice
+        app_log.warning("8: Run_task finished running (run_planned_workflow)")
+
+        try:
+            # When one or more nodes failed in the last iteration, don't iterate further
+            for node_id in nodes:
+                if result_object._get_node_status(DispatchDB()._get_data_store(), node_id) == str(
```

Comment:
Is the dispatcher server no longer going to have the result object in memory during workflow execution?

Reply:
We'll still keep the result object in ram. Some of these statements are unnecessary and we need to take these out, but I need to merge this branch ASAP.